api/server/fntracer.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"strings"
+)
+
+// FnTracer is a custom Tracer which wraps another another tracer
+// its main purpose is to wrap the underlying Span in a FnSpan,
+// which adds some extra behaviour required for sending tracing spans to prometheus
+type FnTracer struct {
+	wrappedTracer opentracing.Tracer
+}
+
+// NewFnTracer returns a new FnTracer which wraps the specified Tracer
+func NewFnTracer(tracerToWrap opentracing.Tracer) opentracing.Tracer {
+	newTracer := &FnTracer{}
+	newTracer.wrappedTracer = tracerToWrap
+	return newTracer
+}
+
+// FnTracer implements opentracing.Tracer
+func (thisFnTracer FnTracer) StartSpan(operationName string, opts ...opentracing.StartSpanOption) opentracing.Span {
+	return NewFnSpan(thisFnTracer.wrappedTracer.StartSpan(operationName, opts...))
+}
+
+// FnTracer implements opentracing.Tracer
+func (thisFnTracer FnTracer) Inject(sm opentracing.SpanContext, format interface{}, carrier interface{}) error {
+	return thisFnTracer.wrappedTracer.Inject(sm, format, carrier)
+
+}
+
+// FnTracer implements opentracing.Tracer
+func (thisFnTracer FnTracer) Extract(format interface{}, carrier interface{}) (opentracing.SpanContext, error) {
+	return thisFnTracer.wrappedTracer.Extract(format, carrier)
+}
+
+// FnSpan is a custom Span that wraps another span
+// which adds some extra behaviour required for sending tracing spans to prometheus
+type FnSpan struct {
+	wrappedSpan opentracing.Span
```

Comment:
same trick here as above and it's more clear we only need to override `Finish*` methods

Reply:
Indeed. Done.